yt/visualization/plot_modifications.py

```diff
@@ -33,8 +33,9 @@
 from yt.visualization._commons import _swap_arg_pair_order, _swap_axes_extents
 from yt.visualization.base_plot_types import CallbackWrapper
 from yt.visualization.image_writer import apply_colormap
+from yt.visualization.plot_window import PWViewerMPL
 
-callback_registry = {}
+callback_registry: Dict[str, Type["PlotCallback"]] = {}
```

Comment:
note that this registry becomes less relevant with this patch, since we don't rely on it to setup annotation methods.
However it's still used in some places (most notably in a docs helper script), so I'm keeping it.